packages/app-config-writer/src/preview-config/index.ts

```diff
@@ -0,0 +1,483 @@
+        await processUi5YamlConfig(fs, basePath, ui5Yaml, '', true);
+        logger?.warn(
+            `UI5 yaml configuration file ${ui5Yaml} it is not being used in any preview script. Outdated preview middleware will be adjusted nevertheless if necessary.`
+        );
+    }
+}
+
+/**
+ * Rename the sandbox file used in the given script.
+ *
+ * @param fs - file system reference
+ * @param basePath - base path to be used for the conversion
+ * @param script - the content of the script
+ * @param logger logger to report info to the user
+ */
+async function renameSandbox(fs: Editor, basePath: string, script: string, logger?: ToolsLogger): Promise<void> {
+    const { path: relativePath } = extractUrlDetails(script);
+    if (relativePath) {
+        const absolutePath = join(await getWebappPath(basePath), relativePath);
+        if (fs.exists(absolutePath)) {
+            fs.move(absolutePath, absolutePath.replace('.html', '_old.html'));
+            logger?.info(renameMessage(relativePath));
+        } else {
+            logger?.warn(`File ${relativePath} not found. Skipping renaming.`);
+        }
+    }
+}
+
+/**
+ * Ensure the @sap/ux-ui5-tooling or @sap-ux/preview-middleware dependency exists in package.json.
+ * If not @sap-ux/preview-middleware will be added as devDependency.
+ *
+ * @param packageJson - the package.json file content
+ * @param fs - file system reference
+ * @param packageJsonPath - the path to the package.json file
+ */
+export function ensurePreviewMiddlewareDependency(
+    packageJson: Package | undefined,
+    fs: Editor,
+    packageJsonPath: string
+): void {
+    if (!packageJson) {
+        return;
+    }
+
+    const hasDependency = (dependency: string): boolean =>
+        !!packageJson?.devDependencies?.[dependency] || !!packageJson?.dependencies?.[dependency];
+
+    const dependencies = ['@sap-ux/preview-middleware', '@sap/ux-ui5-tooling'];
+    if (dependencies.some((dependency) => hasDependency(dependency))) {
+        return;
+    }
+
+    packageJson.devDependencies = { ...packageJson.devDependencies, '@sap-ux/preview-middleware': 'latest' };
+    fs.writeJSON(packageJsonPath, packageJson);
+}
+
+/**
+ * Process the UI5 yaml configuration file.
+ *
+ * @param fs - file system reference
+ * @param basePath - base path to be used for the conversion
+ * @param ui5Yaml - the name of the UI5 yaml configuration file
+ * @param script - the content of the script
+ * @param skipPreviewMiddlewareCreation - (default: true) indicator if the preview middleware creation should be skipped if no preview middleware is configured
+ */
+export async function processUi5YamlConfig(
+    fs: Editor,
+    basePath: string,
+    ui5Yaml: string,
+    script: string,
+    skipPreviewMiddlewareCreation = false
+): Promise<void> {
+    const ui5YamlConfig = await readUi5Yaml(basePath, ui5Yaml, fs);
+    let previewMiddleware = await getPreviewMiddleware(ui5YamlConfig);
```

Comment:
Seems like `ui5-config` does not include comments when loading a specific middleware config 🤷🏻 So writing it after some changes will actually delete the comment even if that part has not been touched at all. @tobiasqueck Is this a bug in `ui5-config`?

Sample:
```yaml
- name: fiori-tools-preview
      afterMiddleware: compression
      configuration:
        libs: true
        debug: true
        adp:
          target:
            url: <some url>
            client: '080'
            authenticationType: reentranceTicket
            # destination: <some name>
          ignoreCertErrors: true
        rta:
          editors:
            - path: /test/adaptation-editor.html
              developerMode: true
```

will become

```yaml
- name: fiori-tools-preview
      afterMiddleware: compression
      configuration:
        libs: true
        debug: true
        flp:
          path: just/a/path/flp.html
        adp:
          target:
            url: <some url>
            client: '080'
            authenticationType: reentranceTicket
          ignoreCertErrors: true
        rta:
          editors:
            - path: /test/adaptation-editor.html
              developerMode: true
```